core/src/test/scala/org/apache/spark/sql/store/SnappyJoinSuite.scala

```diff
@@ -498,8 +501,41 @@ class SnappyJoinSuite extends SnappyFunSuite with BeforeAndAfterAll {
     assert(excatJoinKeys.count() === 500)
   }
 
+  test("SnappyAggregation partitioning") {
+
+    val dimension1 = sc.parallelize(
+      (1 to 1000).map(i => TestDatak(i % 10, i.toString, i % 10)))
+    val refDf = snc.createDataFrame(dimension1)
+    snc.sql("DROP TABLE IF EXISTS PR_TABLE20")
+
+    snc.sql("CREATE TABLE PR_TABLE20(OrderId INT, description String, " +
+        "OrderRef INT) USING column options (" +
+        "PARTITION_BY 'OrderId,OrderRef')")
+    refDf.write.insertInto("PR_TABLE20")
+    val groupBy1 = snc.sql(s"select  OrderRef, orderId from pr_table20 group by OrderRef, orderId")
+    checkForShuffle(groupBy1.logicalPlan, snc, shuffleExpected = false)
+    val groupBy2 = snc.sql(s"select  orderId, sum(orderRef) from pr_table20 group by orderId")
+    checkForShuffle(groupBy2.logicalPlan, snc, shuffleExpected = true)
+  }
+
   def partitionToPartitionJoinAssertions(snc: SnappyContext,
       t1: String, t2: String): Unit = {
+
+    val nullableEquality = snc.sql(s"select P.OrderRef, P.description from " +
+        s"$t1 P JOIN $t2 R ON P.OrderId = R.OrderId" +
+        s" AND P.OrderRef <=> R.OrderRef")
+    // TODO Why an exchange is needed for coalesce.
+    checkForShuffle(nullableEquality.logicalPlan, snc, shuffleExpected = true)
+    assert(nullableEquality.count() === 500)
+
+    val withCoalesce = snc.sql(s"select P.OrderRef, P.description from " +
+        s"$t1 P JOIN $t2 R ON P.OrderId = R.OrderId" +
+        s" AND coalesce(P.OrderRef,0) = coalesce(R.OrderRef,0)")
+    // TODO Why an exchange is needed for coalesce.
```

Comment:
Good question. It should not.

Reply:
The semanticEqual() check for an attribute & its Coalesce returns false.  We need to override semanticEqual method for some of these functions.